pkg/services/searchV2/object/dashboard.go

```diff
@@ -0,0 +1,104 @@
+package object
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strconv"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/searchV2/dslookup"
+	"github.com/grafana/grafana/pkg/services/searchV2/extract"
+)
+
+func NewDashboardObjectReader(lookup dslookup.DatasourceLookup) models.ObjectReader {
```

Comment:
IMO, the goal (if not downright constraint) should be that our schema metadata is good that this can be done largely if not entirely generically/via generated code that calling generic functions.

Reply:
Correct -- the types of things we should pull out for summary *should* be constrained to things you can say by just looking at the document.

This dashboard implementation requires dslookup because existing dashboard JSON has many ways to reference data sources, and we want a single representation to fall out (kind+type+uid).  

NOTE however, that the summary != the object, and the types of fields we want to index/search may change without changing the underlying object schema etc.

In addition to this *static* summary, there will also need to be a set of dynamic properties that do depend on the whole system state.  References have have a bunch of properties like:
1. all references have been checked and are valid
2. some references are invalid
3. all references share the same permissions?

For enterprise, we may need to join additional properties for sorting and reporting like:
* views in last 1,7,30 days
* errors in last 1,7,30 days
* ...

Reply:
Resolving this - we talked it out, and it's pretty clear we can make both a pure declarative/CUE approach and a Go approach live side-by-side. sketch of an approach in [this gist](https://gist.github.com/sdboyer/8a8b1a037926f2407eb2086580a41451)